### PR TITLE
Composer and Bower vendor dirs can be copied from previous release to sp...

### PR DESCRIPTION
**Pull request by rommsen**
...eed up deployment

Added new variables: project_copy_previous_composer_vendors, project_composer_vendor_pathm project_copy_previous_bower_componentsm project_bower_components_path
Added cp tasks
Changed README

### CODE REVIEW

**Comment by ramondelafuente** [comment]
@jaspernbrouwer: Any thoughts on this? Looks ok to me.

File: tasks/main.yml [hunk 1]
```diff
@@ -44,6 +52,14 @@
   environment: project_environment
   when: project_has_npm
 
+- name: Check if bower dir exists
+  stat: path={{ deploy_helper.current_path }}/{{ project_bower_components_path }}
+  register: check_bower_components_path
+
+- name: Copy bower dir if exists to speed up bower install
+  command: /bin/cp -rp {{ deploy_helper.current_path }}/{{ project_bower_components_path }} {{ deploy_helper.new_release_path }}/{{ project_bower_components_path }}
```
Comment (jaspernbrouwer) [inline tasks/main.yml]: Why is `project_bower_components_path` appended to the destination path, while this is not done for composer (on line 42)? Going either way is fine, but lets keep it consistent :smile:

File: tasks/main.yml [hunk 2]
```diff
@@ -34,6 +34,14 @@
   with_items: project_pre_build_commands
   environment: project_environment
 
+- name: Check if vendor dir exists
```
Comment (jaspernbrouwer) [inline tasks/main.yml]: Checking if a previous install exists isn't needed when `project_copy_previous_composer_vendors` is off, so maybe add a `when` to this task?

File: tasks/main.yml [hunk 3]
```diff
@@ -44,6 +52,14 @@
   environment: project_environment
   when: project_has_npm
 
+- name: Check if bower dir exists
```
Comment (jaspernbrouwer) [inline tasks/main.yml]: Checking if a previous install exists isn't needed when `project_copy_previous_bower_components` is off, so maybe add a `when` to this task?

**Comment by jaspernbrouwer** [comment]
There's also NPM in the role, maybe add the same feature for it?

File: tasks/main.yml (hunk 1 above)
Comment (rommsen) [inline tasks/main.yml]: Ok. Will change

File: tasks/main.yml (hunk 3 above)
Comment (rommsen) [inline tasks/main.yml]: yes, you are right

**Comment by rommsen** [comment]
Yeah I dont need npm and I was lazy :D. I will change the PR accordingly

**Comment by jaspernbrouwer** [comment]
Thanx!

**Comment by rommsen** [comment]
I did all the changes wrote some more documentation in the README.

I squashed the whole PR into one commit to have a cleaner history. Hope this is ok.

**Comment by ramondelafuente** [comment]
:+1:

**Pull request merged**